src/uct/api/uct.h

```diff
@@ -636,9 +636,15 @@ void uct_iface_mem_free(const uct_allocated_memory_t *mem);
  * @param [in]  id       Active message id. Must be 0..UCT_AM_ID_MAX-1.
  * @param [in]  cb       Active message callback. NULL to clear.
  * @param [in]  arg      Active message argument.
+ * @param [in]  flags    Required capability flags
  */
+enum {
+    UCT_AM_CB_FLAG_MT_SAFE = UCS_BIT(1), /* callback can be safely called
```

Comment:
`UCT_AM_FLAG_THREAD_SAGE` - "callback can be called from any thread (by default it would be called only a thread which explicitly calls uct_iface_progress"